src/Orchard.Web/Modules/Orchard.MediaLibrary.WebSearch/Controllers/Api/WebSearchApiController.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Web.Http;
+using Orchard.MediaLibrary.WebSearch.Providers;
+using Orchard.Security;
+
+namespace Orchard.MediaLibrary.WebSearch.Controllers.Api {
+    public class WebSearchApiController : ApiController {
+        private readonly IAuthorizer _authorizer;
+        private readonly IEnumerable<IWebSearchProvider> _wsp;
+
+        public WebSearchApiController(IAuthorizer authorizer, IEnumerable<IWebSearchProvider> wsp) {
+            _authorizer = authorizer;
+            _wsp = wsp;
+        }
+
+        [HttpGet]
+        public IHttpActionResult GetImages(string query, string providerType) {
+            if (!_authorizer.Authorize(Permissions.ManageMediaContent)) {
```

Comment:
Let's use a separate, custom permission just for the Web Search instead of ManageMediaContent.